vm/src/obj/objint.rs

```diff
@@ -704,7 +704,7 @@ struct IntOptions {
     #[pyarg(positional_only, optional = true)]
     val_options: OptionalArg<PyObjectRef>,
     #[pyarg(positional_or_keyword, optional = true)]
-    base: OptionalArg<u32>,
+    base: OptionalArg<i32>,
```

Comment:
This doesn't completely fix the problem, it just shifts the range that the problem occurs, because if a PyInt can't fit into a `i32` it still overflows. I recommend changing `base` to be a PyIntRef and passing a `&BigInt` around instead of an `i32`.

Reply:
Thank you for reviewing. I Change base to `PyIntRef`. And i also add test for minus base and Large value than max of i32.